libraries/AP_HAL_ChibiOS/hwdef/JFB100/hwdef.dat

```diff
@@ -0,0 +1,354 @@
+# PG13 SPI6_SCK SPI6
+# PG12 SPI6_MISO SPI6
+# PB5 SPI6_MOSI SPI6
+
+# sensor CS
+PF10 MS5611_CS CS
+PF2  SCHA63T_G_CS CS
+PF3  ICM20602_CS CS SPEED_VERYLOW
+PF4  SCHA63T_A_CS CS
+PF5  FRAM_CS CS SPEED_VERYLOW
+
+# unusued CS pins
+# PF11 SPARE_CS CS (using for WA6)
+PH5  AUXMEM_CS CS
+PI4  EXTERNAL1_CS1 CS
+PI10 EXTERNAL1_CS2 CS
+PI11 EXTERNAL1_CS3 CS
+PI6  EXTERNAL2_CS1 CS
+PI7  EXTERNAL2_CS2 CS
+PI8  EXTERNAL2_CS3 CS
+
+# I2C buses
+
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+
+PH7 I2C3_SCL I2C3
+PH8 I2C3_SDA I2C3
+
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C3 I2C1 I2C2 I2C4
+
+
+# enable pins
+PE3 VDD_3V3_SENSORS_EN OUTPUT HIGH
+
+# start peripheral power off, then enable after init
+# this prevents a problem with radios that use RTS for
+# bootloader hold
+PF12 nVDD_5V_HIPOWER_EN OUTPUT HIGH
+PG4  nVDD_5V_PERIPH_EN OUTPUT HIGH
+
+PG5  VDD_5V_RC_EN OUTPUT HIGH
+PG6  VDD_5V_WIFI_EN OUTPUT HIGH
+PG7  VDD_3V3_SD_CARD_EN OUTPUT HIGH
+
+# drdy pins
+# PE7  DRDY8_NC INPUT (using for WA5)
+PB4  DRDY1_ICM20689 INPUT
+PB14 DRDY2_SCHA63T_GYRO INPUT
+PB15 DRDY3_SCHA63T_ACC INPUT
+PC5  DRDY4_ICM20602 INPUT
+PC13 DRDY5_SCHA63T_GYRO INPUT
+PD10 DRDY6_SCHA63T_ACC INPUT
+PD15 DRDY7_EXTERNAL1 INPUT
+
+# Control of Spektrum power pin
+PE4 SPEKTRUM_PWR OUTPUT HIGH GPIO(73)
+define HAL_GPIO_SPEKTRUM_PWR 73
+
+# Spektrum Power is Active High
+define HAL_SPEKTRUM_PWR_ENABLED 1
+
+# UARTs
+
+# USART2 is telem1
+PD6 USART2_RX USART2
+PD5 USART2_TX USART2
+PD3 USART2_CTS USART2
+PD4 USART2_RTS USART2
+
+# USART1 is GPS1
+PB7 USART1_RX USART1 NODMA
+PB6 USART1_TX USART1 NODMA
+
+# USART3 is telem2
+PD9 USART3_RX USART3
+PD8 USART3_TX USART3
+PD11 USART3_CTS USART3
+PD12 USART3_RTS USART3
+
+# UART4 GPS2
+PD0 UART4_RX UART4 NODMA
+PD1 UART4_TX UART4 NODMA
+
+# USART6 is telem3
+PG9 USART6_RX USART6 NODMA
+# we leave PG14 as an input to prevent it acting as a pullup
+# on the IOMCU SBUS input
+PG14 USART6_TX USART6 NODMA
+# PG15 USART6_CTS USART6 (using for WA8)
+# PG8 USART6_RTS USART6 (using for WA7)
+
+# UART7 is debug
+PF6 UART7_RX UART7 NODMA
+PE8 UART7_TX UART7 NODMA
+
+# UART8 is for IOMCU
+# PE0 UART8_RX UART8 (using for PP3)
+# PE1 UART8_TX UART8 (using for PP5)
+
+# UART for IOMCU
+# IOMCU_UART UART8
+
+# RCInput on the PPM pin, for all protocols
+PI5 TIM8_CH1 TIM8 RCININT PULLUP LOW
+
+# PWM AUX channels
+PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
+PA10 TIM1_CH3 TIM1 PWM(2) GPIO(51)
+PE11 TIM1_CH2 TIM1 PWM(3) GPIO(52)
+PE9  TIM1_CH1 TIM1 PWM(4) GPIO(53)
+PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54)
+PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55)
+# we need to disable DMA on the last 2 FMU channels
+# as timer 12 doesn't have a TIMn_UP DMA option
+PH6  TIM12_CH1 TIM12 PWM(7) GPIO(56) NODMA
+PH9  TIM12_CH2 TIM12 PWM(8) GPIO(57) NODMA
+
+define BOARD_PWM_COUNT_DEFAULT 8
+
+# PWM output for buzzer
+PE5 TIM9_CH1 TIM9 GPIO(77) ALARM
+
+# analog in
+PA0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
```

Comment:
could use ANALOG(14) to align with CubeOrange pins for voltage. Similar for current

Reply:
added ANALOG(14) for voltage and ANALOG(15) for current to align with CubeOrange

Reply:
Actually I got a compiler error, "Error: Duplicate analog pin 14" so I've removed the extra "ANALOG(14)"

Reply:
JFB100 board is using ANALOG(14) (PC4 pin) as SPARE1_ADC1.

Reply:
@jfbblue0922,

Txs for this. The pins numbers are a bit arbitrary really so if possible it is good to align them with existing boards instead of creating new mappings.  It just makes it easier for users to set up the board if we try to keep them consistent.

Maybe we can swap the virtual pin numbers in a follow-up commit.